neptune/new/internal/backends/hosted_neptune_backend.py

```diff
@@ -207,12 +207,20 @@ def get_project(self, project_id: str) -> Project:
                 raise NeptuneLegacyProjectException(project_id)
             return Project(project.id, project.name, project.organizationName)
         except HTTPNotFound:
+            available_workspaces = self.get_available_workspaces()
+
+            if workspace and not any(
+                filter(lambda aw: aw.name == workspace, available_workspaces)
+            ):
+                # Could not found specified workspace, forces listing all projects
+                workspace = None
+            else:
+                available_workspaces = list()
```

Comment:
Why reset this list if workspace is None?
Can we just skip fetching available worksapces?